scripts/jenkins/Jenkinsfile

```diff
@@ -0,0 +1,567 @@
+/* https://build.libelektra.org/jenkins/job/elektra-jenkinsfile/
+ * This file describes how the elektra-jenkinsfile buildjob should be
+ * executed.
+ *
+ * 1. imports and global variables are set
+ * 2. define the main stages of the pipeline
+ * 3. describe sub stages. This is where you will want to add new builds
+ * 4. helper section to help write build scripts
+ */
+
+// FIXME sloc publish errors
+// XXX add missing jobs
+// TODO have a per plugin/binding deps in Dockerfile for easier maintanence
+// TODO add warnings plugins to scan for compiler warnings
+
+// Imports
+import java.text.SimpleDateFormat
+
+// Buildjob properties
+properties([
+    buildDiscarder(logRotator(numToKeepStr: '60', artifactNumToKeepStr: '60'))
+])
+
+// Globals
+docker_node_label = 'docker'
+docker_image_prefix = 'build-elektra-'
+
+DOCKER_IMAGES = [
+    sid: [
+        id: docker_image_prefix + 'debian-sid',
+        context: "./scripts/docker/debian/sid",
+        file: "./scripts/docker/debian/sid/Dockerfile"
+    ],
+    stretch: [
+        id: docker_image_prefix + 'debian-stretch',
+        context: "./scripts/docker/debian/stretch",
+        file: "./scripts/docker/debian/stretch/Dockerfile",
+    ],
+    stretch_minimal: [
+        id: docker_image_prefix + 'debian-stretch-minimal',
+        context: "./scripts/docker/debian/stretch",
+        file: "./scripts/docker/debian/stretch/Dockerfile.minimal",
+    ],
+    stretch_doc: [
+        id: docker_image_prefix + 'debian-stretch-doc',
+        context: "./scripts/docker/debian/stretch",
+        file: "./scripts/docker/debian/stretch/Dockerfile.doc",
+    ],
+    jessie: [
+        id: docker_image_prefix + 'debian-jessie',
+        context: "./scripts/docker/debian/jessie",
+        file: "./scripts/docker/debian/jessie/Dockerfile",
+    ],
+    xenial: [
+        id: docker_image_prefix + 'ubuntu-xenial',
+        context: "./scripts/docker/ubuntu/xenial",
+        file: "./scripts/docker/ubuntu/xenial/Dockerfile",
+    ]
+]
+
+// Define reusable cmake Flag globals
+CMAKE_FLAGS_BASE = [
+    'SITE': '${STAGE_NAME}',
+    'KDB_DB_SYSTEM': '${HOME}/.config/kdb/system',
+    'KDB_DB_SPEC': '${HOME}/.config/kdb/spec',
+    'CMAKE_INSTALL_PREFIX': '${WORKSPACE}/system',
+    'INSTALL_SYSTEM_FILES': 'OFF',
+    'BUILD_DOCUMENTATION': 'OFF'
+]
+
+CMAKE_FLAGS_BUILD_ALL = [
+    'BINDINGS': 'ALL;EXPERIMENTAL;-DEPRECATED',
+    'PLUGINS': 'ALL;EXPERIMENTAL;-DEPRECATED',
+    'TOOLS': 'ALL'
+]
+
+CMAKE_FLAGS_COVERAGE = ['ENABLE_COVERAGE': 'ON']
+
+CMAKE_FLAGS_JAVA = [
+    'JAVA_AWT_INCLUDE_PATH': '/usr/lib/jvm/java-8-openjdk-amd64/include/'
```

Comment:
Isn't this dependent on the docker image?

Reply:
kind of. It is a monkey patch kind of solution anyway as cmake should be able to find it without providing the path explicitly but it does not.

Reply:
Yeah, Debian loves shipping CMake files that are unable to find its own files, already happened multiple times (e.g. python or swig3).

Forking all these CMake files, however, puts quite a maintenance burden on us. For Java we actually nearly included it but we decided to avoid forking CMake Modules.

Now the scripts/configure- contains all our workarounds, maybe you want to use them, too?

Reply:
there is nothing java related in the configure scripts anymore (only commented out section about setting JAVA_HOME, which I already do in the docker image)

Reply:
cmakes unsets the var because it can not find `jawt.h`. The file is not available in stretch https://packages.debian.org/stretch/amd64/openjdk-8-jdk-headless/filelist which seems like a bug as it is available in the i386 package https://packages.debian.org/stretch/i386/openjdk-8-jdk/filelist ....